src/System.IO.Compression/tests/DeflateStreamTests.cs

```diff
@@ -512,6 +546,92 @@ public static IEnumerable<object[]> RoundtripCompressDecompressOuterData
             Assert.Equal<byte>(data, decompressed.ToArray());
         }
 
+        [OuterLoop]
+        [Theory]
+        [MemberData("RoundtripCompressDecompressOuterData")]
+        public async Task RoundTripWithFlush(bool useAsync, bool useGzip, int chunkSize, int totalSize, CompressionLevel level)
+        {
+            byte[] data = new byte[totalSize];
+            new Random(42).NextBytes(data);
+
+            var compressed = new MemoryStream();
+            var compressor = useGzip ? (Stream)new GZipStream(compressed, level, true) : new DeflateStream(compressed, level, true);
+            {
+                for (int i = 0; i < data.Length; i += chunkSize) // not using CopyTo{Async} due to optimizations in MemoryStream's implementation that avoid what we're trying to test
+                {
+                    switch (useAsync)
+                    {
+                        case true: await compressor.WriteAsync(data, i, chunkSize); break;
+                        case false: compressor.Write(data, i, chunkSize); break;
+                    }
+                }
+            }
+            switch (useAsync)
+            {
+                case true: await compressor.FlushAsync(); break;
+                case false: compressor.Flush(); break;
+            }
```

Comment:
The ```WriteAfterFlushing``` test performs a series of writes with flushes and validation tests in between calls - is that what you mean by "in the middle" or were you thinking something else?

I do not have any tests for flushing at the beginning. I will add one.